assets/course-theme/blocks/register-template-blocks.js

```diff
@@ -36,8 +36,6 @@ export function registerTemplateBlocks( blocks ) {
 			return;
 		}
 
-		const isTemplate =
-			'lesson' === postType && editPost.isEditingTemplate();
-		toggleBlockRegistration( isTemplate );
+		toggleBlockRegistration( false );
```

Comment:
With the current version of Gutenberg, this has the same effect as the previous code.

See https://github.com/WordPress/gutenberg/blob/060d4ca57ab8441255db34e308c21a640adad2a6/packages/edit-post/src/store/selectors.js#L534-L540

Basically, it was checking `'lesson' === postType && 'wp_template' === postType`.

Checking the history of that function, it had some variations, but this should be enough. I tested it in the lesson editor and in the template using WP 6.4 and 6.6, and it seems to be working well (blocks available on the template and not available on the lesson editor).